tests/PHPStan/Rules/Methods/data/static-has-method.php

```diff
@@ -0,0 +1,18 @@
+<?php declare(strict_types = 1);
+
+namespace StaticHasMethodCall;
+
+class rex_var {}
+
+class HelloWorld
+{
+	public function sayHello(): void
+	{
+		if (!method_exists(rex_var::class, 'varsIterator')) {
+			return;
+		}
+
+		$it = rex_var::varsIterator();
```

Comment:
the `NodeScopeResolverTest` seems to work as expected with this PR.

what is missing is a proper fix in `StaticMethodCallCheck` so no error is emitted for this line.
was not yet able to figure out how this can work